SpokestackTests/TFLiteKeywordRecognizerTest.swift

```diff
@@ -0,0 +1,197 @@
+        XCTAssertEqual(filterResult, MockKeywordModels.filterOutput)
+        
+        // encode
+        let encoder = try! Interpreter(modelPath: MockKeywordModels.encodePath)
+        try! encoder.allocateTensors()
+        _ = MockKeywordModels.encodeInput.withUnsafeBytes { try! encoder.copy(Data($0), toInputAt: 0) }
+        _ = MockKeywordModels.input.withUnsafeBytes { try! encoder.copy(Data($0), toInputAt: 1) }
+        _ = try! encoder.invoke()
+        let encoderOutput = try! encoder.output(at: MockKeywordModels.validIndex)
+        XCTAssertEqual(encoderOutput.data, MockKeywordModels.encodeOutputData)
+        let encoderResult = encoderOutput.data.toArray(type: Int32.self, count: encoderOutput.data.count/4)
+        XCTAssertEqual(encoderResult, MockKeywordModels.encodeOutput)
+        
+        // detect
+        let detect = try! Interpreter(modelPath: MockKeywordModels.detectPath)
+        try! detect.allocateTensors()
+        _ = MockKeywordModels.detectInput.withUnsafeBytes { try! detect.copy(Data($0), toInputAt: 0) }
+        _ = try! detect.invoke()
+        let detectOutput = try! detect.output(at: MockKeywordModels.validIndex)
+        XCTAssertEqual(detectOutput.data, MockKeywordModels.detectOutputData)
+        let detectResult = detectOutput.data.toArray(type: Int32.self, count: detectOutput.data.count/4)
+        XCTAssertEqual(detectResult, MockKeywordModels.detectOutput)
+    }
+    
+    func testStartStop() {
+        self.context!.isActive = false
+        self.recognizer?.startStreaming()
+        self.recognizer?.stopStreaming()
+    }
+    
+    func testProcess() {
+        // setup
+        self.recognizer?.context.addListener(self.delegate)
+        self.recognizer?.context.isActive = true
+        self.recognizer?.context.isSpeech = true
+        self.recognizer?.startStreaming()
+        let recognizeExpectation = XCTestExpectation(description: "process without failure.")
+        let timeoutExpectation = XCTestExpectation(description: "process without failure.")
+        let failureExpectation = XCTestExpectation(description: "process without failure.")
+        self.delegate.asyncExpectation = failureExpectation
+        self.delegate.didRecognizeExepctation = recognizeExpectation
+        self.delegate.didTimeoutExpectation = timeoutExpectation
+        
+        // timeout
+        for _ in 0...1 {
+            self.recognizer?.process(Frame.voice(frameWidth: 20, sampleRate: 16000))
+        }
+        self.context?.isActive = false
+        wait(for: [timeoutExpectation], timeout: 1)
+        XCTAssert(self.delegate.timedOut)
+        XCTAssertFalse(self.delegate.didError)
+        XCTAssertFalse(self.delegate.recognized)
+        
+        // don't process if the pipeline isn't active
+        delegate.reset()
+        self.recognizer?.startStreaming()
+        self.recognizer?.context.isSpeech = true
```

Comment:
Why set to true then immediately to false?

Reply:
this is a test of the vad edge logic